src/graph/graphcontext.c

```diff
@@ -620,42 +620,41 @@ Index GraphContext_GetIndex
 // create an exact match index for the given label and attribute
 bool GraphContext_AddExactMatchIndex
 (
-	Index *idx,                 // [input/output] index created
-	GraphContext *gc,           // graph context
-	SchemaType schema_type,     // type of entities to index nodes/edges
-	const char *label,          // label of indexed entities
-	const char **fields_str,    // fields to index
-	uint fields_count,          // number of fields to index
-	bool should_reply           // should reply to client
+	Index *idx,              // [input/output] index created
+	GraphContext *gc,        // graph context
+	SchemaType schema_type,  // type of entities to index nodes/edges
+	const char *label,       // label of indexed entities
+	const char **attrs,      // attributes to index
+	uint attrs_count         // number of attributes to index
 ) {
-	ASSERT(idx    !=  NULL);
-	ASSERT(gc     !=  NULL);
-	ASSERT(label  !=  NULL);
-	ASSERT(fields_str !=  NULL);
-	ASSERT(fields_count > 0);
+	ASSERT(idx   !=  NULL);
+	ASSERT(gc    !=  NULL);
+	ASSERT(label !=  NULL);
+	ASSERT(attrs !=  NULL);
+	ASSERT(attrs_count > 0);
 
 	// retrieve the schema for this label
 	bool      index_changed = false;
 	Schema    *s            = GraphContext_GetSchema(gc, label, schema_type);
-	ResultSet *result_set   = should_reply ? QueryCtx_GetResultSet() : NULL;
+	ResultSet *result_set   = QueryCtx_GetResultSet();
 
 	if(s == NULL) {
 		// schema doesn't exists, create it
 		s = GraphContext_AddSchema(gc, label, schema_type);
 	}
 
-	for(uint i = 0; i < fields_count; i++) {
+	for(uint i = 0; i < attrs_count; i++) {
 		// create index field
-		const char *field = fields_str[i];
-		IndexField idx_field;
-		Attribute_ID f_id = GraphContext_FindOrAddAttribute(gc, field, NULL);
-		IndexField_Default(&idx_field, f_id, field);
-		if(Schema_AddIndex(idx, s, &idx_field, IDX_EXACT_MATCH) == INDEX_OK) {
+		IndexField field;
+		const char *attr = attrs[i];
+		Attribute_ID attr_id = GraphContext_FindOrAddAttribute(gc, attr, NULL);
+
+		IndexField_NewExactMatchField(&field, attr, attr_id);
+
+		if(Schema_AddIndex(idx, s, &field, IDX_EXACT_MATCH) == INDEX_OK) {
 			index_changed = true;
-			if(should_reply) {
-				// update result-set
-				ResultSet_IndexCreated(result_set, INDEX_OK);
-			}
+			// update result-set
+			ResultSet_IndexCreated(result_set, INDEX_OK);
```

Comment:
maybe move this to line 666?

Reply:
The function `ResultSet_IndexCreated` counts how many "indices" were created.
In the case of: `CREATE INDEX FOR (n:N) ON (n.a, n.b)` we report 2 indices were created, one for each attribute
if we move this call outside of the loop our statistics would be incorrect.